alpakka/src/main/scala/org/scanamo/ScanamoAlpakka.scala

```diff
@@ -12,7 +12,7 @@ import scala.concurrent.{ExecutionContext, Future}
 object ScanamoAlpakka {
   import cats.instances.future._
 
-  def exec[A](client: DynamoClient)(op: ScanamoOps[A])(implicit ec: ExecutionContext): Future[A] =
-    op.foldMap(AlpakkaInterpreter.future(client)(ec))
+  def exec[A](client: DynamoClient)(op: ScanamoOps[A])(implicit ec: ExecutionContext, retrySettings : RetrySettings): Future[A] =
```

Comment:
I think the retry policy should be provided explicitly by the user, we want them to make an informed choice about that. So let's make it an explicit parameter everywhere.

Reply:
Defining the retry policy as explicitly here it's not the good choice according to me. As we define it explicitly then whenever anybody makes the call has to pass the retry policy which reduces the Scanamo clarity for the queries.

I personally choose it as implicit so that the user just needs to define only once and then use of every query. Even if we make it implicit then also give the user the informed choice about that because the user has to define its policy (of implicit type) in the scope to run the query.

Reply:
You are completely right 👍 In fact, we can say the same about the `client`. Look at the `foldMap`, we create a new natural transformation every time we call `exec`, even though it will always be the same! I think all these parameters should bubble up to `ScanamoAlpakka`:
```scala
class ScanamoAlpakka(
  client: DynamoClient,
  retryPolicy : RetryPolicy = RetryPolicy.default
)(implicit ec: ExecutionContext) extends ScanamoOpsA ~> Future {
  final def exec[A](op: ScanamoOps[A]): Future[A] = apply(op)

  final override def apply[A](ops: ScanamoOpsA[A]) = ...
}
```

I'm happy to do that in a separate PR as it is a wholesome change to all the interpreters, so I'm fine with the retry being an implicit for now.